pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -309,18 +318,37 @@ static char* _check_pyarray_rank(PyArrayObject *a, int rank, bool allow_empty)
  */
 static char* _check_pyarray_order(PyArrayObject *a, int flag)
 {
-	if (flag == NO_ORDER_CHECK)
-		return NULL;
+    if (flag == NO_ORDER_CHECK)
+        return NULL;
 
-	if (!PyArray_CHKFLAGS(a, flag))
-	{
-		char order = (flag == NPY_ARRAY_C_CONTIGUOUS ? 'C' : (flag == NPY_ARRAY_F_CONTIGUOUS ? 'F' : '?'));
+    bool valid = true;
+    if (flag == NPY_ARRAY_C_CONTIGUOUS) {
+        int nd = PyArray_NDIM(a);
+        npy_intp* np_strides = PyArray_STRIDES(a);
+        for (int i = 1; i<nd; ++i) {
+            valid = valid & (np_strides[i-1] >= np_strides[i]);
+        }
+    }
+    else if (flag == NPY_ARRAY_F_CONTIGUOUS) {
+        int nd = PyArray_NDIM(a);
+        npy_intp* np_strides = PyArray_STRIDES(a);
+        for (int i = 1; i<nd; ++i) {
+            valid = valid & (np_strides[i-1] <= np_strides[i]);
+        }
+    }
+    else {
+        valid = PyArray_CHKFLAGS(a, flag);
+    }
```

Comment:
What flags can this function process? All those handled by [`PyArray_CHKFLAGS`](https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_CHKFLAGS) or just some of them?

Reply:
All of them, but I don't know what is available and I don't think we currently use this after this PR

Reply:
Comment modified : [4f14854](https://github.com/pyccel/pyccel/pull/2477/commits/4f14854103f2c19950d916b275906e2e75c83b7c)